### PR TITLE
Add support for Godot 4.2 typed for loop iterator variable, fixes #236

### LINKED ISSUE #236: Add support for Godot 4.2 static typing for for loop variable
Godot 4.2 now supports typing the variable declaration in a for loop:
https://github.com/godotengine/godot/pull/80247

However gdformat fails when this syntax is used.

As of Godot 4.2 Beta, this syntax is valid:
```
var dict := {1: 2}
for key: String in dict:
    print(key)
```

However, running gdformat on it results in this error:
`Previous tokens: [Token('NAME', 'key')]`

### PR DESCRIPTION
**Pull request by neilgoodman**
I tested the formatter locally using Godot 4.2 beta 2 on one of my projects in addition to the new unit tests.

### CODE REVIEW

File: gdtoolkit/common/ast.py
Comment (Scony) [inline gdtoolkit/common/ast.py:1]: could you please add `# pylint: disable=too-many-branches` before `def _load_sub_statements(self):` ? It looks like we got too many branches right now, but this pylint check is irrelevant here.

**Review by Scony — CHANGES_REQUESTED** [review]
Very well PR with full test coverage - loving it!

File: gdtoolkit/common/ast.py
Comment (neilgoodman) [reply gdtoolkit/common/ast.py:1]: Done!

**Review by Scony — APPROVED** [review]
Thanks!

**Pull request merged**